src/features/definitionProvider.ts

```diff
@@ -29,11 +29,23 @@ export default class CSharpDefinitionProvider extends AbstractSupport implements
 
         return serverUtils.goToDefinition(this._server, req, token).then(gotoDefinitionResponse => {
 
+            // the defintion is in source
             if (gotoDefinitionResponse && gotoDefinitionResponse.FileName) {
+
+                // if it is part of an already used metadata file, retrieve its uri instead of going to the physical file
+                if (gotoDefinitionResponse.FileName.startsWith("$metadata$")) {
```

Comment:
We should add something to the OmniSharp response to indicate that the result is in metadata, rather than checking the file name. Not a blocking a issue, but I wanted to capture it.

Reply:
yes I agree, good point